go/host/l1/publisher.go

```diff
@@ -170,6 +173,23 @@ func (p *Publisher) ExtractSecretResponses(block *types.Block) []*ethadapter.L1R
 	return secretRespTxs
 }
 
+func (p *Publisher) FetchLatestSeqNo() (*big.Int, error) {
```

Comment:
good idea to read the last sequence number from the actual contract, as the ultimate source of truth